### PR TITLE
fix: DidDocument default context

### LINKED ISSUE #4501: Unexpected context for DID documents
# Bug Report

## Describe the Bug
When performing a did:web resolution call towards Identity Hub it returns a DID document with `https://w3id.org/did-resolution/v1` context but expected it to be: `https://www.w3.org/ns/did/v1`.

### Expected Behavior
When returning a DID document it is expected that it follows the Decentralized Identifiers (DIDs) v1.0 specification.

### Observed Behavior
The current `https://w3id.org/did-resolution/v1` context as specified https://github.com/eclipse-edc/Connector/blob/main/spi/common/identity-did-spi/src/main/java/org/eclipse/edc/iam/did/spi/document/DidDocument.java#L31 is for "Decentralized Identifier Resolution (DID Resolution)" see https://w3c.github.io/did-resolution/#example

## Steps to Reproduce

1. Start MVD sample
2. Query did document with `curl http://localhost:7093/.well-known/did.json`. 
3. Observe that following document is returned
```
{
  "service": [
    {
      "id": "provider-credentialservice-1",
      "type": "CredentialService",
      "serviceEndpoint": "http://localhost:7091/api/presentation/v1/participants/ZGlkOndlYjpsb2NhbGhvc3QlM0E3MDkz"
    },
    {
      "id": "provider-catalogserver-dsp",
      "type": "ProtocolEndpoint",
      "serviceEndpoint": "http://localhost:8092/api/dsp"
    }
  ],
  "verificationMethod": [
    ...
  ],
  "authentication": [],
  "id": "did:web:localhost%3A7093",
  "@context": [
    "https://w3id.org/did-resolution/v1"
  ]
}
```

## Context Information
EDC version 0.10.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by perher**
## What this PR changes/adds

Changes the default json-ld context for `DidDocument`.

## Why it does that

To follow the "Decentralized Identifiers (DIDs) v1.0 specification".

## Linked Issue(s)

Closes #4501

### CODE REVIEW

File: spi/common/identity-did-spi/src/main/java/org/eclipse/edc/iam/did/spi/document/DidDocument.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 @JsonDeserialize(builder = DidDocument.Builder.class)
 public class DidDocument {
 
-    public static final String DID_RESOLUTION_CONTEXT = "https://w3id.org/did-resolution/v1";
+    public static final String DID_RESOLUTION_CONTEXT = "https://www.w3.org/ns/did/v1";
```
Comment (wolf4ood) [inline spi/common/identity-did-spi/src/main/java/org/eclipse/edc/iam/did/spi/document/DidDocument.java:31]:
```suggestion
    public static final String DID_CONTEXT = "https://www.w3.org/ns/did/v1";
```
Just a nit but i would change also the constant name

**Comment by wolf4ood** [comment]
also @perher  if you rebase the PR it should fix the failing checks 

Thanks

**Comment by perher** [comment]
I've rebased and renamed constant now

**Comment by perher** [comment]
rebased, but not passing the ECA check now for some reason.. will retry again later

**Comment by perher** [comment]
updated DEPENDENCIES file

**Comment by perher** [comment]
@wolf4ood I was already on latest main, but updated the DEPENDENCY file again according to checker

**Comment by perher** [comment]
another dependency issue, I need help from maintainers now, please let me know what to do or feel free to push to my branch

**Review by wolf4ood — APPROVED** [review]

**Comment by wolf4ood** [comment]
Thanks @perher

**Pull request merged**